newrelic/terraform-provider-newrelic
# feat: Add evaluation delay to nrql condition

# Description

Adds `evaluation_delay` to NRQL Alert Condition resource. Enables users to manage their evaluation delay on NRQL Alert Conditions

**Depends on the New Relic Go client PR here https://github.com/newrelic/newrelic-client-go/pull/977** 

Fixes [NR-58050](https://issues.newrelic.com/browse/NR-58050)

## Type of change

Please delete options that are not relevant.


## Checklist:

Please delete options that are not relevant.


## How to test this change?

Using the Go client changes in https://issues.newrelic.com/browse/NR-58050, create a NRQL Alert Condition with evaluation delay set. 

Additionally, ensure your account is added to the `DCON/evaluation_delay_agql` feature flag. 

```
variable "nr_api_key" {
  type = string
  description = "your new relic API key"
}

variable "account_id" {
  type = number
  description = "your new relic account"
}

variable "region" {
  type = string
  description = "the region you're using."
}

provider newrelic {
  api_key = var.nr_api_key
  account_id = var.account_id
  region = var.region
}
resource "newrelic_nrql_alert_condition" "your-condition" {
  account_id =<Your Account Id>
  policy_id = <Your Policy Id>
  type = "static"
  name = "Large NRQL Query"
  description = <<EOT
  Some description
  EOT
  runbook_url = "https://newrelic.com"
  enabled = false
  violation_time_limit_seconds = 28800

  nrql {
    query = "SELECT count(message) FROM Log where message RLIKE r'^(\\w+\\s+\\w+\\s+[\\w,:]+)(.*)kernel:(.*):(.*)Test\\sLink\\sis\\sDown'"
  }

  critical {
    operator = "above"
    threshold = 1
    threshold_duration = 300
    threshold_occurrences = "all"
  }

  warning {
    operator = "below"
    threshold = 4000
    threshold_duration = 3000
    threshold_occurrences = "all"
  }
  fill_option = "static"
  fill_value = 5
  aggregation_window = 540
  aggregation_method = "cadence"
  aggregation_delay = 1200
  slide_by = 30
  expiration_duration = 900
  open_violation_on_expiration = true
  close_violations_on_expiration = true
  evaluation_delay = 600
}
```
